iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -346,7 +346,16 @@ public Map<String, String> getBasicStatistics(Partish partish) {
               stats.put(StatsSetupConst.NUM_FILES, summary.get(SnapshotSummary.TOTAL_DATA_FILES_PROP));
             }
             if (summary.containsKey(SnapshotSummary.TOTAL_RECORDS_PROP)) {
-              stats.put(StatsSetupConst.ROW_COUNT, summary.get(SnapshotSummary.TOTAL_RECORDS_PROP));
+              long totalRecords = Long.parseLong(summary.get(SnapshotSummary.TOTAL_RECORDS_PROP));
+              if (summary.containsKey(SnapshotSummary.TOTAL_EQ_DELETES_PROP) &&
+                  summary.containsKey(SnapshotSummary.TOTAL_POS_DELETES_PROP)) {
+                Long actualRecords =
+                    totalRecords - (Long.parseLong(summary.get(SnapshotSummary.TOTAL_EQ_DELETES_PROP)) >
```

Comment:
It is a code smell a bit imho.

When you are using conditional operator your eyes need to parse a variable that you put it into 2 methods. So to read / debug the value you need to have an understanding what the summary.get returns and after the Long.prase (last is easier for sure.  Maybe It is fully OK and you can choose to not accept this opinion but I think it is nicer if you create a variable for those values.

Reply:
Btw why this getBasicStatistics needs switch case?

Reply:
Fixed. 

getBasicStatistics doesn't need a switch case anymore.
Thanks! :)